azurerm/internal/services/media/media_job_resource_test.go

```diff
@@ -0,0 +1,250 @@
+			Check: resource.ComposeAggregateTestCheckFunc(
+				check.That(data.ResourceName).Key("name").HasValue("Job-1"),
+				check.That(data.ResourceName).Key("description").HasValue("Job description"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: resource.ComposeAggregateTestCheckFunc(
+				check.That(data.ResourceName).Key("description").HasValue("Updated description"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeAggregateTestCheckFunc(
+				check.That(data.ResourceName).Key("description").HasValue("Job description"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (MediaJobResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	id, err := parse.JobID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Media.JobsClient.Get(ctx, id.ResourceGroup, id.MediaserviceName, id.TransformName, id.Name)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving Job %s (Media Services Account %s) (resource group: %s): %v", id.Name, id.MediaserviceName, id.ResourceGroup, err)
+	}
+
+	return utils.Bool(resp.JobProperties != nil), nil
+}
+
+func (r MediaJobResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_media_job" "test" {
+  name                        = "Job-1"
+  resource_group_name         = azurerm_resource_group.test.name
+  media_services_account_name = azurerm_media_services_account.test.name
+  transform_name              = azurerm_media_transform.test.name
+  description                 = "Job description"
+  priority                    = "Normal"
+  input_asset {
+    asset_name = azurerm_media_asset.input.name
+  }
+  output_asset {
+    asset_name = azurerm_media_asset.output.name
+  }
+}
+
+`, r.template(data))
+}
+
+func (r MediaJobResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_media_job" "import" {
+  name                        = azurerm_media_job.test.name
+  resource_group_name         = azurerm_media_job.test.resource_group_name
+  media_services_account_name = azurerm_media_job.test.media_services_account_name
+}
+
+`, r.basic(data))
+}
+
+func (r MediaJobResource) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_media_job" "test" {
+  name                        = "Job-1"
+  resource_group_name         = azurerm_resource_group.test.name
+  media_services_account_name = azurerm_media_services_account.test.name
+  transform_name              = azurerm_media_transform.test.name
+  description                 = "Updated description"
+  priority                    = "Low"
+  input_asset {
+    asset_name = azurerm_media_asset.input.name
+  }
+  output_asset {
+    asset_name = azurerm_media_asset.output.name
+  }
+}
+
+`, r.template(data))
+}
+
+func (r MediaJobResource) label(data acceptance.TestData) string {
```

Comment:
and then we can remove this test ans the complete and update tests will test this